word_segmentation.py

```diff
@@ -286,35 +293,287 @@ def preprocess():
+                print("LINE {} -  ICU BIES STRING : {}".format(line_counter, icu_bies_str))
+                print('**********************************************************************************')
+            line_counter += 1
+    graph_clust_freq = grapheme_clusters_dic
+    graph_clust_freq = {k: v for k, v in sorted(graph_clust_freq.items(), key=lambda item: item[1], reverse=True)}
+    graph_clust_ratio = graph_clust_freq
+    total = sum(graph_clust_ratio.values(), 0.0)
+    graph_clust_ratio = {k: v / total for k, v in graph_clust_ratio.items()}
+    return graph_clust_ratio
+
+
+def add_additional_bars(read_filename, write_filename):
+    """
+    This function reads a segmented file and add bars around each space in it. It assumes that spaces are used as
+    breakpoints in the segmentation (just like "|")
+    Args:
+        read_filename: Address of the input file
+        write_filename: Address of the output file
+    """
+    wfile = open(write_filename, 'w')
+    with open(read_filename) as f:
+        for line in f:
+            line = line.strip()
+            if len(line) == 0:
+                continue
+            new_line = ""
+            for i in range(len(line)):
+                ch = line[i]
+                if ch == " ":
+                # If you want to put lines bars around punctuations as well, you should use comment previous if and
+                # uncomment the next if.
+                # The following if will put bars for !? as |!||?| instead of |!|?|. This should be fixed if the
+                # following if is going to be used. It can easily be fixed by keeping track of the last character in
+                # new_line.
+                # if 32 <= ord(ch) <= 47 or 58 <= ord(ch) <= 64:
+                    if i == 0:
+                        if i+1 < len(line) and line[i+1] == "|":
+                            new_line = new_line + "|" + ch
+                        else:
+                            new_line = new_line + "|" + ch + "|"
+                    elif i == len(line)-1:
+                        if line[i-1] == "|":
+                            new_line = new_line + ch + "|"
+                        else:
+                            new_line = new_line + "|" + ch + "|"
+                    else:
+                        if line[i-1] != "|" and line[i+1] != "|":
+                            new_line = new_line + "|" + ch + "|"
+                        if line[i-1] == "|" and line[i+1] != "|":
+                            new_line = new_line + ch + "|"
+                        if line[i-1] != "|" and line[i+1] == "|":
+                            new_line = new_line + "|" + ch
+                        if line[i-1] == "|" and line[i+1] == "|":
+                            new_line = new_line + ch
+                else:
+                    new_line += ch
+            new_line += "\n"
+            wfile.write(new_line)
+
+
+def compute_ICU_accuracy(filename):
```

Comment:
I think you have the ICU BIES code at least 3 times in this file: in preprocess_Thai, preprocess_Burmese, and here in compute_ICU_accuracy.  Consider refactoring the code to reduce duplication.